File: Notes/lib/data/net/api/api_service.dart
import 'package:dio/dio.dart';
import 'package:notes/data/common/response_common.dart';
import 'package:notes/screen/note/model/list_note_model.dart';
import 'package:notes/utils/constants.dart';
import 'package:retrofit/http.dart';

part 'api_service.g.dart';

const BASE_URL = "https://ktor-project-ducdv.herokuapp.com/";
const HEADER_KEY = "Authorization";
const CONTENT_TYPE = "application/json";

@RestApi(baseUrl: BASE_URL)
abstract class ApiClientNote {
  factory ApiClientNote(Dio dio) = _ApiClientNote;

  @POST("/login")
  Future<ResponseLogin> logIn(@Field(USERNAME) String username,
      @Field(PASSWORD) String password,);

  @GET("/notes")
  Future<ListNoteModel> getAllDataNotes(@Header(HEADER_KEY) String auth,);
}

File: Notes2/app/src/main/java/vn/deviot/notes/data/api/ApiService.kt
package vn.deviot.notes.data.api

import okhttp3.RequestBody
import retrofit2.http.*
import vn.deviot.notes.data.common.ResponseData
import vn.deviot.notes.screen.login.model.LoginRp
import vn.deviot.notes.screen.notes.model.ResponseNotes
import vn.deviot.notes.utils.AUTHORIZATION
import vn.deviot.notes.utils.NOTE

interface ApiService {

    @POST("login")
    suspend fun logIn(
        @Body body: RequestBody
    ): ResponseData<LoginRp>?

    @GET("notes")
    suspend fun getAllNote(
        @Header(AUTHORIZATION) Authorization: String
    ): ResponseData<ResponseNotes>?

    @FormUrlEncoded
    @POST("add/note")
    suspend fun addNote(
        @Header(AUTHORIZATION) Authorization: String,
        @Field(NOTE) note: String
    ): ResponseData<String?>?
}